### ISSUE TITLE
Invalid layer breaks reported for BD.

### ISSUE THREAD

**Issue by Didrole** [superg/redumper#176]
The `redumper_dump_dvd` outputs invalid layer breaks to the log.  

eg:
```disc structure:
  DI { format: 1, layer: 0, sequence number: 0, unit size: 32, continuation: 0 }
    disc type: BDO
    disc size: 120mm
    disc class: 0
    disc version: 1
    layers count: 2
    channel length: 74.5nm
    polarity: 3
    BCA: no
    PSN last { PSN: 30545951, hPSN: 0x1D2181F }
    AUN { PSN: [1048576 .. 13267966], length: 12219392, hPSN: [0x100000 .. 0xCA73FE] }
  DI { format: 1, layer: 1, sequence number: 1, unit size: 32, continuation: 0 }
    disc type: BDO
    disc size: 120mm
    disc class: 0
    disc version: 1
    layers count: 2
    channel length: 74.5nm
    polarity: 3
    BCA: no
    PSN last { PSN: 30545951, hPSN: 0x1D2181F }
    AUN { PSN: [20286464 .. 32505854], length: 12219392, hPSN: [0x1358C00 .. 0x1EFFFFE] }

layer break: 2524995655
```

After checking the source code, it appears the line https://github.com/superg/redumper/blob/99ab2005036d8c32682af6b624b9cf79218140c1/dvd/dvd_dump.ixx#L466 mistakenly makes a copy of the `READ_DISC_STRUCTURE_DiscInformationUnit` structure, which would work if the `body` field wasn't a zero-length array.  
Due to this all accesses to the `body` field of the copy result in out of bound memory access.

**Comment by Didrole** [comment]
Please check again, I may have explained the situation badly.  
The `unit` variable should be a reference or a pointer here, because it's not the `body` field is garbage.

**Comment by superg** [comment]
Yes I apologize, I've been driving in my car and thinking about it!
The problem is not the body member access but rather that payload is not getting copied by value, this is really good catch, it should be auto &.

**Comment by superg** [comment]
It used to be access by the pointer but I really like the construct:
`auto &a = (T &)b[c]`
better vs
`auto a = (T *)&b[c]`
And using the reference instead of pointer.

**Comment by Didrole** [comment]
Thank you for checking again.  
Closing this ticket, as #177 has been merged.
